Simplify the selfhosted AU guide (#40054)
gravitational/teleport: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx

the next major version (adding a new APT/YUM/zypper repo for `stable/v14`).
Alternatively, you can use the `stable/rolling` channel, which contains
Teleport v13.3.2 forward, including future major releases.
</Admonition>

## Prerequisites

- Familiarity with the [Upgrading Compatibility Overview](./overview.mdx) guide,
  which describes the sequence in which to upgrade components of your cluster.
- Self-hosted Teleport cluster running at least v14.3.7 or v15.1.3. 
- (!docs/pages/includes/tctl-tsh-prerequisite.mdx!)
- (!docs/pages/includes/tctl.mdx!)

## Step 1/4. Check the version advertised by the Teleport Proxy Service

The Teleport Proxy service exposes by default a version channel called `default`
that serves the same version the Proxy Service is running.

> If your proxy public address is <Var value="teleport.example.com:443" />, you can

Reviewer: ```suggestion
If your proxy public address is <Var name="teleport.example.com:443" />, you can
```

query the version server with the following command:

```code

> $ curl "https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version"

Reviewer: ```suggestion
$ curl "https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version"
```

(=teleport.version=)
```

<Details title="Changing which version is served by the version server">

By default, the version server has a single `default` channel, serving the
version of the Teleport Proxy Service. If you want to override the default version or add other channels you
can use the `automatic_upgrades_channels` field in the Proxy Service configuration file:

```yaml
proxy_service:
  enabled: "yes"
  automatic_upgrades_channels:
    # Override the default version channel reachable at

>     # https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version

Reviewer: ```suggestion
    # https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version
```

    default:
      static_version: v14.2.1
    # Define a new version channel with a static version reachable at

>     # https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/m-static-channel/version

Reviewer: ```suggestion
    # https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/m-static-channel/version
```

    my-static-channel:
      static_version: v14.2.0
    # Define a new version channel forwarding requests to an upstream version server
    my-remote-channel:
      forward_url: https://updates.releases.teleport.dev/v1/stable/cloud
```

> You must ensure all proxies share the same `automatic_upgrades_channels`

Reviewer: ```suggestion
You must ensure all Proxy Service instances share the same `automatic_upgrades_channels`
```

> configuration. If proxies are configured differently, you will experience agents

Reviewer: ```suggestion
configuration. If Proxy Service instances are configured differently, you will experience agents
```

flickering between versions as the version served is not consistent across

> proxies.

Reviewer: ```suggestion
instances.
```

the next major version (adding a new APT/YUM/zypper repo for `stable/v14`).
Alternatively, you can use the `stable/rolling` channel, which contains
Teleport v13.3.2 forward, including future major releases.
</Admonition>

## Prerequisites

- Familiarity with the [Upgrading Compatibility Overview](./overview.mdx) guide,
  which describes the sequence in which to upgrade components of your cluster.
- Self-hosted Teleport cluster running at least v14.3.7 or v15.1.3. 
- (!docs/pages/includes/tctl-tsh-prerequisite.mdx!)
- (!docs/pages/includes/tctl.mdx!)

> <Admonition type="note">

Reviewer: I think this admonition would make more sense in the changelog. Users reading this guide won't see instructions to set up a public web server. We could also provide a brief architectural overview in the introduction of this page that indicates that the Proxy Service hosts the version server. This way, we don't risk adding noise for users with an admonition.

Author: I removed the note, we already have a changelog entry for this

   An earlier version of this guide required you to setup a public web server to
   host the Teleport version, this is not required anymore as the Teleport proxy
   service can now act as a version server.
</Admonition>

- (!docs/pages/includes/tctl.mdx!)

> ## Step 1/4. Check the version served by the proxy

Reviewer: I would avoid using "the proxy" if we're referring to the Teleport Proxy Service. This (a) keeps terms consistent in the docs and (b) distinguishes the Teleport Proxy Service from other proxy servers running in the user's infrastructure.

> The Teleport Proxy serviceexposes by default a version channel called `default`

Reviewer: ```suggestion
The Teleport Proxy service exposes by default a version channel called `default`
```

> and serving the same version the proxy is running.

Reviewer: ```suggestion
that serves the same version the Proxy Service is running.
```

If your proxy public address is <Var value="teleport.example.com:443" />, you can

> query the version server with the following command:\

Reviewer: ```suggestion
query the version server with the following command:
```

```code
$ curl "https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version"
(=teleport.version=)
```

<Details title="Changing which version is served by the version server">

> By default, the version server has a single `default` channel, serving the proxy
> version. If you want to override the default version or add other channels you
> can pass custom proxy configuration:

Reviewer: ```suggestion
By default, the version server has a single `default` channel, serving the
version of the Teleport Proxy Service. If you want to override the default version or add other channels you
can use the `automatic_upgrades_channels` field in the Proxy Service configuration file:
```

```yaml
proxy_service:
  enabled: "yes"
  automatic_upgrades_channels:
    # Override the default version channel
    default:
      static_version: v14.2.1
    # Define a new version channel with a static version

>     additional/channel/static:

Reviewer: So any `/`-separated string key defines a route on the version server? Which path would the user expect to query here (is there a prefix for the path?)?

Author: I added a URL example and removed the slashes as it was confusing. We support slashes in the channel name, but they're not mandatory.

   to the right release channel:
   
   ```code

>    $ echo "<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default" | sudo tee /etc/teleport-upgrade.d/endpoint

Reviewer: ```suggestion
   $ echo "<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default" | sudo tee /etc/teleport-upgrade.d/endpoint
```
